metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -412,3 +412,85 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
 
     def get_report(self) -> StaleEntityRemovalSourceReport:
         return self.report
+
+
+class PresetConfig(StatefulIngestionConfigBase, ConfigModel):
+    # See the Preset.io API docs: https://api-docs.preset.io/
+    manager_uri: str = Field(
+        default="https://api.app.preset.io/", description="Preset.io API URL"
+    )
+    connect_uri: str = Field(default=None, description="Preset workspace URL.")
+    display_uri: Optional[str] = Field(
+        default=None,
+        description="optional URL to use in links (if `connect_uri` is only for ingestion)",
+    )
+    api_key: Optional[str] = Field(default=None, description="Preset.io API key.")
+    api_secret: Optional[str] = Field(default=None, description="Preset.io API secret.")
+
+    # Configuration for stateful ingestion
+    stateful_ingestion: Optional[StatefulStaleMetadataRemovalConfig] = Field(
+        default=None, description="Superset Stateful Ingestion Config."
+    )
+
+    options: Dict = Field(default={}, description="")
+    env: str = Field(
+        default=DEFAULT_ENV,
+        description="Environment to use in namespace when constructing URNs",
+    )
+    database_alias: Dict[str, str] = Field(
+        default={},
+        description="Can be used to change mapping for database names in superset to what you have in datahub",
+    )
+
+    @validator("connect_uri", "display_uri")
+    def remove_trailing_slash(cls, v):
+        return config_clean.remove_trailing_slashes(v)
+
+    @root_validator
+    def default_display_uri_to_connect_uri(cls, values):
+        base = values.get("display_uri")
+        if base is None:
+            values["display_uri"] = values.get("connect_uri")
+        return values
+
+
+@platform_name("Preset")
+@config_class(PresetConfig)
+@support_status(SupportStatus.TESTING)
+@capability(
+    SourceCapability.DELETION_DETECTION, "Optionally enabled via stateful_ingestion"
+)
+class PresetSource(SupersetSource):
+    """
+    Variation of the Superset plugin that works with Preset.io (Apache Superset SaaS).
+    """
+
+    config: PresetConfig
+    platform = "preset"
+
+    def __init__(self, ctx: PipelineContext, config: SupersetConfig):
+        StatefulIngestionSourceBase.__init__(config, ctx)
+        self.config = config
+        self.report = StaleEntityRemovalSourceReport()
+
+        login_response = requests.post(
```

Comment:
Can we refractor get token piece into separate `get_access_token` method in SupersetSource and only override `create` and `get_access_token` method in PresetSource ?